src/tasks/trainerUtils.py

```diff
@@ -227,180 +156,287 @@ def parse_ev_script_file(file_path):
+            trainers.append(trainer)
+        return trainers
+    if "Celebi" in lookup:
+        temp_trainer_IDs = parse_random_teams(file_path, "", team_num, lookup)
+        for ID in temp_trainer_IDs:
+            trainer = get_single_trainer(zoneID, ID, temp_trainer_IDs, lookup)
+            trainers.append(trainer)
+        return trainers
+
+def get_assorted_trainer_data(file_path, areaName, zoneID, trainerID1, trainerID2, args):
+    trainer = []
+    trainers = []
+    cyrus_lookup = f"ev_{areaName.lower()}_randomteam_cyrus"
+    master_trainer_types = ["fire", "water", "electric", "grass", "ice", "fighting", "poison", "ground", "flying", "psychic", "bug", "rock", "ghost", "dragon", "dark", "steel", "fairy"]
+    starters = ["piplup", "turtwig" ,"chimchar"]
+    for starter in starters:
+        rival_lookup = f"ev_{areaName.lower()}_randomteam_barry_{starter}"
+        rival_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, rival_lookup, 4)
+        for rival in rival_teams:
+            trainer = rival
+            trainers.append(trainer)
+    if trainer == []:
+        cyrus_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, cyrus_lookup, 4)
+        for cyrus in cyrus_teams:
+            trainer = cyrus
+            trainers.append(trainer)
+    if trainer == []:
+        master_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, "Masters", master_trainer_types)
+        for master in master_teams:
+            trainer = master
+            trainers.append(trainer)
+    if trainer == []:
+        celebi_teams = get_random_team_data(file_path, areaName, zoneID, trainerID1, trainerID2, "Celebi", 7)
+        for celebi_team in celebi_teams:
+            trainer = celebi_team
+            trainers.append(trainer)
+    if trainer == []:
+        print("Lucas and Dawn's Single Battles are not yet supported:", areaName, args[0])
+        '''
+        ### This section is currently in the progress of being improved and finalized in ev_script.
+
+        for support in ["lucas", "dawn"]:
+            temp_support_IDs = parse_random_teams(file_path, f"ev_{areaName.lower()}_support_{support}", 3, None)
+            for ID in temp_support_IDs:
+                trainer = diff_trainer_data(None, zoneID, int(ID))
+                trainer["name"] = f"{trainer['name']} {mon} Team {str(temp_rival_IDs.index(ID) + 1)}"
+                trainer["format"] = "Single"
+                trainer["link"] = ""
+                trainers.append(trainer)
+        '''
+    return trainers
+
 def get_support_trainers_data(file_path, area_name, support_name, zoneID):
     """
     The purpose of this function is to get the support trainers data
     """
     temp_support_IDs = []
+    bad_support_lookup1 = f"ev_r207_func_17" ### These bad lookups are for Lucas and Dawn on Route 207
```

Comment:
same as earlier, constants go to the top, and should be in full caps to indicate what they are